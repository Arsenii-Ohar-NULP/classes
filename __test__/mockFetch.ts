export const mockFetch = () => {
    global.fetch = jest.fn();
}