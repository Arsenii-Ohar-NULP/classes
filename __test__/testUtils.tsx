import React, { PropsWithChildren } from 'react';
import { render, renderHook } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import type { PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import type { RootState, AppStore } from 'pages/redux/store';
// As a basic setup, import your same slice reducers
import { authInitialState, authSlice } from 'pages/redux/auth';
import { classSlice, classesInitialState } from 'pages/redux/classes';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderHookWithProviders(
  ui,
  {
    preloadedState = { auth: authInitialState, classes: classesInitialState },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { auth: authSlice.reducer, classes: classSlice.reducer },
      preloadedState,
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...renderHook(() => ui(), { wrapper: Wrapper, ...renderOptions }) };
}
