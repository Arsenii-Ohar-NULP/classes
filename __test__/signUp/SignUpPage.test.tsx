import * as React from 'react';

import {renderWithProviders} from '__test__/testUtils';
import SignUp from 'app/auth/signUp/page';

const navigateMock = jest.fn();
jest.mock('next/navigation', () => (
    {
        useRouter: () => ({
            push: navigateMock,
            replace: navigateMock
        }),
        useSearchParams: () => ({
            username: "arseniiohar"
        })
    }
));
describe('sign up page', () => {
    it('should match a snapshot', () => {
        const page = renderWithProviders(<SignUp/>);
        expect(page.container).toMatchSnapshot();
    })
})