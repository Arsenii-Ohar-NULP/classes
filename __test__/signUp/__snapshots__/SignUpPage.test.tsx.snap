// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sign up page should match a snapshot 1`] = `
<div>
  <div>
    <div
      class="container w-100 h-100 d-relative"
    >
      <div
        class="grid p-3"
      >
        <div
          class="d-flex align-middle"
        >
          <h1
            class="p-3"
          >
            Sign up
          </h1>
          <hr />
        </div>
        <div
          class="row"
        >
          <form
            class="needs-validation grid gy-3"
            id="form"
            name="form"
          >
            <div
              class="row"
            >
              <div
                class="p-1 "
              >
                <label
                  class="form-label"
                  for="Username"
                >
                  <b>
                    Username
                  </b>
                </label>
                <input
                  class="form-control"
                  id="Username"
                  name="Username"
                  placeholder="Enter your username"
                  value="arseniiohar"
                />
                <div
                  class="invalid"
                />
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="p-1 "
              >
                <label
                  class="form-label"
                  for="Password"
                >
                  <b>
                    Password
                  </b>
                </label>
                <input
                  class="form-control"
                  id="Password"
                  name="Password"
                  placeholder="Enter your password"
                  type="password"
                  value=""
                />
                <div
                  class="invalid"
                />
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="p-1 col-lg"
              >
                <label
                  class="form-label"
                  for="FirstName"
                >
                  <b>
                    First Name
                  </b>
                </label>
                <input
                  class="form-control"
                  id="FirstName"
                  name="FirstName"
                  placeholder="Enter your first name"
                  value=""
                />
                <div
                  class="invalid"
                />
              </div>
              <div
                class="p-1 col-lg"
              >
                <label
                  class="form-label"
                  for="LastName"
                >
                  <b>
                    Last Name
                  </b>
                </label>
                <input
                  class="form-control"
                  id="LastName"
                  name="LastName"
                  placeholder="Enter your last name"
                  value=""
                />
                <div
                  class="invalid"
                />
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="p-1 col-lg"
              >
                <label
                  class="form-label"
                  for="Email"
                >
                  <b>
                    Email
                  </b>
                </label>
                <input
                  class="form-control"
                  id="Email"
                  name="Email"
                  placeholder="Enter your email"
                  value=""
                />
                <div
                  class="invalid"
                />
              </div>
              <div
                class="p-1 col-lg"
              >
                <label
                  class="form-label"
                  for="PhoneNumber"
                >
                  <b>
                    Phone Number
                  </b>
                </label>
                <input
                  class="form-control"
                  id="PhoneNumber"
                  name="PhoneNumber"
                  placeholder="Enter your phone number"
                  value=""
                />
                <div
                  class="invalid"
                />
              </div>
            </div>
            <div
              class="invalid"
            />
            <hr />
            <div
              class="container d-flex gap-2 justify-content-center mx-auto my-3 text-center p-3 start-0"
            >
              <button
                class="btn btn-primary rounded-4 p-2 px-4"
                id="signUpButton"
                type="button"
              >
                Sign up
              </button>
              <div
                class="vr my-1"
              />
              <button
                class="btn btn-secondary rounded-4 px-3"
                type="button"
              >
                Log in
              </button>
            </div>
          </form>
        </div>
      </div>
      <footer
        class="d-flex justify-content-center p-3 align-items-end"
      >
        <a
          class="text-decoration-none"
          href="example.com"
        >
          <div
            class="d-flex justify-content-center align-items-center gap-1 bg-dark p-2 rounded"
          >
            <img
              alt="Classes Logo"
              class="border rounded-3 p-1 m-0 fill-light classes-pic"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <h3
              class="m-0 align-middle text-primary"
            >
              Classes
            </h3>
          </div>
        </a>
      </footer>
    </div>
  </div>
</div>
`;
